### ISSUE TITLE
Support for .M3U8 video files in the enclosure.

### ISSUE THREAD

**Issue by dhk2** [Heckie75/kodi-addon-podcast#25]
PeerTube podcast RSS feeds use .m3u8 file for the video version in the enclosure tag. These fail to load in Kodi when using the Podcast Plugin. From what I can tell Kodi supports m3u8 data but not the file extension or some such issue. This should hopefully be easy to resolve within the plugin somehow.

example RSS feed with the issue:    

https://p2ptube.us/feeds/podcast/videos.xml?videoChannelId=19086

My PeerTube plugin generates an audio only version with an mp4 file in the enclosure and it works fine.

https://p2ptube.us/plugins/podcast2/router/podcast2?channel=otra

**Comment by Heckie75** [comment]
@dhk2 
I've checked https://p2ptube.us/feeds/podcast/videos.xml?videoChannelId=19086

Since you've said that Kodi supports m3u but not m3u8 I've changed the file extentions of enclosures. But also then Kodi hangs. There's no error, but it just displays a spinning wheel and nothing happens. 

What is your expectation? It would be possible to fallback to the ```audioenclosure``` element in case that the normal enclosure links to a ressource ending with m3u8 and (of course) there is an ```audioenclosure```?

**Comment by dhk2** [comment]
yeah, the audioenclosure was an artifact of my first attempts to be backwardly compatible with podcast apps that don't support m3u8, removed that, thanks for noticing.

With my plugin I've worked around the issue currently by making a customized RSS feed with a video version using the mp4 directly in the enclosure, but will only work with instances that install my plugin. In addition, moving forward more Podcast hosts are going to be using m3u8 and adding support shouldn't hurt.

This thread on Reddit suggests the way to get it to work with Kodi is to create a text file containing the path to the .m3u8 file and having a .m3u or .strm extension. Don't know if that's relevant to the way your plug-in works.

https://www.reddit.com/r/Addons4Kodi/comments/ohgg7m/question_regarding_kodi_playing_m3u8_files/

Thanks for looking into this
